run_unittests.py

```diff
@@ -4189,7 +4189,7 @@ def test_dependency(self):
             raise unittest.SkipTest('zlib not found with pkg-config')
         a = (("dependency('zlib', method : 'fail')", "'fail' is invalid"),
              ("dependency('zlib', static : '1')", "[Ss]tatic.*boolean"),
-             ("dependency('zlib', version : 1)", "[Vv]ersion.*string or list"),
+             ("dependency('zlib', version : 1)", "Item must be a list or one of <class 'str'>"),
```

Comment:
That's the poor message from stringlistify() used everywhere. I agree it can be improved, but it's not specific to this case.

Reply:
You have the file:line:col information, so that shouldn't be too hard to fix for the user tbh.

Reply:
i can't imagine that the users gets the column of the version parameter. Because there is not really a data path for that to show up. Is there any reason why you changed the previous code to stringlistify?
Sorry for beeing picky about error messages, but users really need those to be actionable to have a positive experience with meson.